src/Controller/AbstractController.php

```diff
@@ -113,9 +113,7 @@ public function dispatch(Request $request, Response $response = null)
           ->setResponse($response)
           ->setTarget($this);
 
-        $result = $this->getEventManager()->trigger(MvcEvent::EVENT_DISPATCH, $e, function ($test) {
-            return ($test instanceof Response);
-        });
+        $result = $this->getEventManager()->trigger(MvcEvent::EVENT_DISPATCH, $e, false);
```

Comment:
The third parameter of the `trigger` method must be `null` or `callable` (see [`Zend\EventManager\EventManager::trigger`](https://github.com/zendframework/zend-eventmanager/blob/d8899a5867f981229a19794b790c33fdc32d5c59/src/EventManager.php#L167-L177)). Your code change breaks several tests.

Perhaps you can create a unit test and show what you are trying to fix?